### PR TITLE
added script to roll and sign tarballs

### PR DESCRIPTION
**Pull request by marcusmueller**

### CODE REVIEW

**Comment by noc0lour** [comment]
I talked with @marcusmueller today and one of the problems of my script was the inability of `git archive` to pull `volk` into the tarball.

blake2/sha512 hashes were an idea before I talked to Maitland where he pointed me to the fact that `gpg2` is already hashing & signing if you run `--detach-sign`. 
Actually the file with the list of hashes is basically worthless, since you can verify the integrity with the `.asc` file, I personally wouldn't verify integrity with an unsigned list of hashes.

**Comment by dkozel** [comment]
I defer to @noc0lour and @marcusmueller on the details.

We could sign the hash list and it would be an improvement over this, but having an unsigned list of hashes of signed files is sufficient to detect if either file changes and to verify the GNU Radio source right?

I vote to merge this and iterate afterwards.

**Comment by noc0lour** [comment]
The `.asc` file already contains a hash to verify the corresponding tarball AND is cryptographically signed so you can verify @marcusmueller actually created the tarball & hash. So what's the benefit of the hash list at all? I'm just curious why we are even creating this list?

**Comment by dkozel** [comment]
I see your point. The hashes file lets you quickly verify file integrity (but not authenticity) without having gpg setup.

File: dtools/bin/roll-tarball.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+plaindirpath=${tempdir}/${plaindir}
+
+mkdir "${plaindirpath}"
+mkdir -p "${archivedir}"
+
+pwd
+echo "Checking out git revision ${gitversion} into folder ${plaindirpath}/…"
+pushd "${plaindirpath}"
+git --git-dir="${gitroot}/.git" checkout --recurse-submodules -f ${gitversion}
+#git --git-dir="${gitroot}/.git" --work-tree=. submodule update --init volk
```
[the remark this answers is not in the document]
Comment (marcusmueller) [reply dtools/bin/roll-tarball.sh]: absolutely – I've just removed this, it was but a remnant of an older attempt.

**Comment by noc0lour** [comment]
@dkozel I've tried to extract the digest without using gpg, and that's not really feasible. So probably having the checksum in a separate file might be good enough for some (who trust the internet)'

**Comment by marcusmueller** [comment]
@noc0lour well the chain of trust is, assuming integrity of all cryptographic software involved

 letsencrypt -> GR web server -> GR web server dudes -> some random weirdo maintainer dude

instead of

 some random weirdo who acts like a maintainer dude

My guess is that the former is only marginally better than the second for quite a lot of people. Not that I agree the least.

**Pull request merged**